python-checks/src/test/resources/checks/unusedLocalVariable.py

```diff
@@ -144,3 +144,17 @@ def foo[T](a: T) -> T:
         ...
 
     return foo
+
+def additional_test_for_loop_comprehension():
+    return ["a" for i in range(5)] # Noncompliant
```

Comment:
Since we changed the issue message for comprehension indices, I think it would make sense to have an assertion on the message as well.